_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw_par.cpp

```diff
@@ -2002,6 +2025,14 @@ mfxStatus CheckAndFixCtrl(
         }
     }
 
+    mfxExtVP9Segmentation* seg_video = GetExtBuffer(video);
+    if (seg_video)
+    {
+        const mfxExtVP9Param& extParSeg = GetExtBufferRef(video);
+        mfxInfoMFX video_par = video.mfx;
+        checkSts = CheckSegmentationParam(*seg_video, extParSeg.FrameWidth, extParSeg.FrameHeight, caps, video_par, &ctrl);
```

Comment:
what if ```checkSts``` was set to some warning before? You will overwrite it with ```MFX_ERR_NONE```. I think you need to do following thing:
```
sts = CheckSegmentationParam(*seg_video, extParSeg.FrameWidth, extParSeg.FrameHeight, caps, video_par, &ctrl);
MFX_CHECK_STS(sts);
```